time stamp
---
bool is_target
